crates/oxc_linter/src/rules/jest/max_nested_describe.rs

```diff
@@ -0,0 +1,397 @@
+    ///             describe('qux', () => {
+    ///                 describe('quxx', () => {
+    ///                     describe('too many', () => {
+    ///                         it('should get something', () => {
+    ///                             expect(getSomething()).toBe('Something');
+    ///                         });
+    ///                     });
+    ///                 });
+    ///             });
+    ///         });
+    ///     });
+    /// });
+    ///
+    /// describe('foo', function () {
+    ///     describe('bar', function () {
+    ///         describe('baz', function () {
+    ///             describe('qux', function () {
+    ///                 describe('quxx', function () {
+    ///                     describe('too many', function () {
+    ///                         it('should get something', () => {
+    ///                             expect(getSomething()).toBe('Something');
+    ///                         });
+    ///                     });
+    ///                 });
+    ///             });
+    ///         });
+    ///     });
+    /// });
+    ///
+    /// // valid
+    /// describe('foo', () => {
+    ///     describe('bar', () => {
+    ///         it('should get something', () => {
+    ///             expect(getSomething()).toBe('Something');
+    ///         });
+    ///     });
+    ///     describe('qux', () => {
+    ///         it('should get something', () => {
+    ///             expect(getSomething()).toBe('Something');
+    ///         });
+    ///     });
+    /// });
+    ///
+    /// describe('foo2', function () {
+    ///     it('should get something', () => {
+    ///         expect(getSomething()).toBe('Something');
+    ///     });
+    /// });
+    ///
+    /// describe('foo', function () {
+    ///     describe('bar', function () {
+    ///         describe('baz', function () {
+    ///             describe('qux', function () {
+    ///                 describe('this is the limit', function () {
+    ///                     it('should get something', () => {
+    ///                         expect(getSomething()).toBe('Something');
+    ///                     });
+    ///                 });
+    ///             });
+    ///         });
+    ///     });
+    /// });
+    /// ```
+    ///
+    MaxNestedDescribe,
+    style,
+);
+
+impl Rule for MaxNestedDescribe {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let max = value
+            .get(0)
+            .and_then(|config| config.get("max"))
+            .and_then(serde_json::Value::as_number)
+            .and_then(serde_json::Number::as_u64)
+            .map_or(5, |v| usize::try_from(v).unwrap_or(5));
+
+        Self { max }
+    }
+
+    fn run_once(&self, ctx: &LintContext) {
+        let mut describes_hooks_depth: Vec<ScopeId> = vec![];
+        let mut possibles_jest_nodes = collect_possible_jest_call_node(ctx);
+        possibles_jest_nodes.sort_by_key(|n| n.node.id());
```

Comment:
What's the benefit of sorting here? Is it worth the extra runtime overhead it adds?

Reply:
It's more easy to handle the iteration, or it will need to check its node.id and parent recursively.